src/controller/timeline-controller.ts

```diff
@@ -78,7 +88,18 @@ class TimelineController extends EventHandler {
       ranges.push([startTime, endTime]);
     }
 
-    this.Cues.newCue(this.captionsTracks[trackName], startTime, endTime, screen);
+    if (this.config.renderNatively) {
+      this.Cues.newCue(this.captionsTracks[trackName], startTime, endTime, screen);
+    } else {
+      const fakeTrack = {
+        cues: [],
+        addCue (cue) {
+          this.cues.pusj(cue);
```

Comment:
Typo?
```suggestion
          this.cues.push(cue);
```
Didn’t typescript complain about the invalid method call?

Reply:
No because `private Cues: any`. Thanks for the heads up. I'll address the typo and typing.

Reply:
Didn’t even look at the typings as I assumed it was array of cue object🧐
I just skimmed through the phone for now, I’ll give more thorough look once I get back to the office.

Reply:
@OrenMe I addressed this issue by requiring the cues handler to return an array, and also accept null for the textTrack argument. Since the default behavior is to use native tracks this is backwards compatible. It's also a lot cleaner than what I was doing above even without the typo! https://github.com/video-dev/hls.js/pull/2550/commits/272e1d060215906c9342f5c6a9dd013b15f1ccf6